diff --git a/lib/iruby/command.rb b/lib/iruby/command.rb
index 52c998d..38bb2f2 100644
--- a/lib/iruby/command.rb
+++ b/lib/iruby/command.rb
@@ -104,7 +104,7 @@ Try `ipython help` for more information.
       Dir.chdir(working_dir) if working_dir
 
       require boot_file if boot_file
-      check_bundler {|e| IRuby.logger.warn "Could not load bundler: #{e.message}" }
+      # check_bundler {|e| IRuby.logger.warn "Could not load bundler: #{e.message}" }
 
       require 'iruby'
       Kernel.new(config_file).run
@@ -129,7 +129,7 @@ Try `ipython help` for more information.
 
       check_version
       check_registered_kernel
-      check_bundler {|e| STDERR.puts "Could not load bundler: #{e.message}" }
+      # check_bundler {|e| STDERR.puts "Could not load bundler: #{e.message}" }
 
       Kernel.exec('ipython', *@args)
     end
@@ -143,15 +143,15 @@ This might not work. Run 'iruby register --force' to fix it." if @iruby_path !=
       end
     end
 
-    def check_bundler
-      require 'bundler'
-      raise %q{iruby is missing from Gemfile. This might not work.
-Add `gem 'iruby'` to your Gemfile to fix it.} unless Bundler.definition.specs.any? {|s| s.name == 'iruby' }
-      Bundler.setup
-    rescue LoadError
-    rescue Exception => e
-      yield(e)
-    end
+#     def check_bundler
+#       # require 'bundler'
+#       # raise %q{iruby is missing from Gemfile. This might not work.
+# # Add `gem 'iruby'` to your Gemfile to fix it.} unless Bundler.definition.specs.any? {|s| s.name == 'iruby' }
+#       # Bundler.setup
+#     rescue LoadError
+#     rescue Exception => e
+#       yield(e)
+#     end
 
     def register_kernel(force_p=false)
       if force_p
